src/Database/Pgsql/Driver/Pgsql.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the tenancy/tenancy package.
+ *
+ * Copyright Tenancy for Laravel
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @see https://tenancy.dev
+ * @see https://github.com/tenancy
+ */
+
+namespace Tenancy\Database\Drivers\Pgsql\Driver;
+
+use Illuminate\Database\ConnectionInterface;
+use Illuminate\Support\Facades\App;
+use Illuminate\Support\Facades\DB;
+use Tenancy\Database\Drivers\Pgsql\Concerns\ManagesSystemConnection;
+use Tenancy\Hooks\Database\Contracts\ProvidesDatabase;
+use Tenancy\Hooks\Database\Events\Drivers as Events;
+use Tenancy\Hooks\Database\Support\QueryManager;
+use Tenancy\Identification\Contracts\Tenant;
+
+class Pgsql implements ProvidesDatabase
+{
+    protected $queryManager;
+
+    public function __construct()
+    {
+        $this->queryManager = App::make(QueryManager::class);
+    }
+
+    public function configure(Tenant $tenant): array
+    {
+        $config = [];
+
+        event(new Events\Configuring($tenant, $config, $this));
+
+        return $config;
+    }
+
+    public function create(Tenant $tenant): bool
+    {
+        $config = $this->configure($tenant);
+
+        event(new Events\Creating($tenant, $config, $this));
+
+        $connection = $this->system($tenant);
+
+        $result = $this->queryManager->setConnection($connection)
+                     ->process(function () use ($config) {
+                         $this->statement("CREATE USER \"{$config['username']}\" WITH PASSWORD '{$config['password']}'");
+                         if ($config['schema'] == 'public') {
+                             $this->statement("CREATE DATABASE \"{$config['database']}\" WITH OWNER = \"{$config['username']}\"");
+                         } else {
+                             $this->statement("CREATE SCHEMA IF NOT EXISTS \"{$config['schema']}\" AUTHORIZATION \"{$config['username']}\"");
```

Comment:
Cant you simply always run this? When it already exists, it won't create it anyways 😅

Reply:
For create, yes.  For modify, probably, though it's a bit of unnecessary overhead.  For delete, absolutely not.  This way it follows a consistent pattern

Reply:
Let's give this some thought today, maybe there's a better way to approach this part of it ;)

Reply:
Can't we run `DROP SCHEMA IF EXISTS RESTRICT`? That should prevent deletion when it's not empty ;)

Reply:
But we don't guarantee all tables/views/etc are deleted before deleting a tenant, and RESTRICT will prevent schema deletion if they're not.  I _REALLY_ don't like that approach.

Other than "it's just not pretty enough", do you have any real objections to this code....? I'll gladly take "functional" over "clever" any day

Reply:
Checked back on this again, I still feel like RESTRICT is the right call here. RESTRICT is also the default in PSQL.

Reply:
Can you clarify what you mean by "is the default"?

Reply:
We are not checking in the MySQL driver whether tables are empty before deleting, so why would we start doing it here.